src/java/org/apache/cassandra/cql3/UpdateParameters.java

```diff
@@ -163,8 +166,21 @@ public Cell<?> addCell(ColumnMetadata column, ByteBuffer value) throws InvalidRe
 
     public Cell<?> addCell(ColumnMetadata column, CellPath path, ByteBuffer value) throws InvalidRequestException
     {
+        // General column value size
         Guardrails.columnValueSize.guard(value.remaining(), column.name.toString(), false, clientState);
 
+        if (column.type instanceof AsciiType) // Ascii size specific guardrail
```

Comment:
nit: prefer comparing the enum type, instead of `instanceof`, e.g.

```suggestion
        if (column.type.asCQL3Type() == CQL3Type.Native.ASCII)
```